test.disabled/ejabberd_tests/tests/rest_client_SUITE.erl

```diff
@@ -521,3 +555,198 @@ assert_json_room_sse_message(Expected, Received) ->
+            escalus:send(Alice, escalus_stanza:roster_add_contact(Bob,
+                         [], <<"Bob">>)),
+            PushReqB = escalus:wait_for_stanza(Alice),
+            escalus:assert(is_roster_set, PushReqB),
+            escalus:send(Alice, escalus_stanza:iq_result(PushReqB)),
+            escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice)),
+            %% Alice sends subscribed presence
+            escalus:send(Alice,
+                         escalus_stanza:presence_direct(
+                             escalus_client:short_jid(Bob),
+                             <<"subscribed">>)),
+            % now check Bob's roster
+            timer:sleep(100),
+            {?OK, R4} = gett("/contacts", BCred),
+            Result4 = decode_maplist(R4),
+            [Res4] = Result4,
+            #{jid := AliceJID, subscription := <<"to">>,
+                ask := <<"none">>} = Res4,
+            ok
+        end
+    ),
+    ok.
+
+add_contact_and_be_invited(Config) ->
+    escalus:fresh_story(
+        Config, [{alice, 1}, {bob, 1}],
+        fun(Alice, Bob) ->
+            AliceJID = escalus_utils:jid_to_lower(
+                escalus_client:short_jid(Alice)),
+            BCred = credentials({bob, Bob}),
+            % bob has empty roster
+            {?OK, R} = gett("/contacts", BCred),
+            Res = decode_maplist(R),
+            [] = Res,
+            % adds Alice
+            AddContact = #{jid => AliceJID},
+            {?NOCONTENT, _} = post(<<"/contacts">>, AddContact,
+                                   BCred),
+            % and she is in his roster, with empty status
+            {?OK, R2} = gett("/contacts", BCred),
+            Result = decode_maplist(R2),
+            [Res2] = Result,
+            #{jid := AliceJID, subscription := <<"none">>,
+              ask := <<"none">>} = Res2,
+            % and he received a roster push
+            Push = escalus:wait_for_stanza(Bob),
+            escalus:assert(is_roster_set, Push),
+            %% she adds him and invites
+            escalus:send(Alice, escalus_stanza:roster_add_contact(Bob,
+                         [],
+                         <<"Bobek">>)),
+            escalus:assert_many([is_roster_set, is_iq_result],
+                                escalus:wait_for_stanzas(Alice, 2)),
+            escalus:send(Alice,
+                         escalus_stanza:presence_direct(
+                             escalus_client:short_jid(Bob),
+                             <<"subscribe">>)),
+            escalus:assert(is_roster_set, escalus:wait_for_stanza(Alice)),
+            escalus:assert(is_presence_with_type, [<<"subscribe">>],
+                           escalus:wait_for_stanza(Bob, 1)),
+            % now check Bob's roster, and it is empty...
+            {?OK, R4} = gett("/contacts", BCred),
+            Result4 = decode_maplist(R4),
+            [] = Result4,
+            % because it is stated in RFC3921, 8.2.6, and implemented
```

Comment:
What exactly is the question you would like to ask? The RFC is pretty straightforward. I think the logic behind it is that the {none, in} is a temporary state meant to handle a situation whereby someone you don't know sends you a subscribe request while you are offline. The roster item is created so that the server can send you the subscription later on - you either accept the subscription or reject it, in both cases moving the roster item out of this state. If you had created the item before, then it indeed disappears from your roster, but instead you see the subscription request. Come to think of it, it makes sense, though in REST API it gets a bit counterintuitive. Anyway, the RFC says "should", so we can change it or make it configurable.